graphbolt/src/partitioned_cache_policy.cc

```diff
@@ -0,0 +1,214 @@
+/**
+ *   Copyright (c) 2023, GT-TDAlab (Muhammed Fatih Balin & Umit V. Catalyurek)
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * @file partitioned_cache_policy.cc
+ * @brief Partitioned cache policy implementation on the CPU.
+ */
+#include "./partitioned_cache_policy.h"
+
+#include <numeric>
+
+namespace graphbolt {
+namespace storage {
+
+constexpr int grain_size = 64;
+
+template <typename BaseCachePolicy>
+PartitionedCachePolicy<BaseCachePolicy>::PartitionedCachePolicy(
+    int64_t capacity, int64_t num_partitions)
+    : capacity_(capacity) {
+  for (int64_t i = 0; i < num_partitions; i++) {
+    const auto begin = i * capacity / num_partitions;
+    const auto end = (i + 1) * capacity / num_partitions;
+    policies_.emplace_back(end - begin);
```

Comment:
This line of code assumes BaseCachePolicy's constructer only take 1 integer as input, which is not explicit. Maybe add clear definition of BaseCachePolicy in cache_policy.h.

Reply:
Can we do this when we add another caching policy? I feel that it is better to refactor when the need arises.

Reply:
I added a definition of BaseCachePolicy however there is no way to force derived classes to constructors. You can only force member functions.